scipy/stats/_hypotests.py

```diff
@@ -965,24 +958,24 @@ def _binomial_maximisation_of_p_value_with_nuisance_param(
         # This operation compensate numerical errors because sums of
         # p_values_arr should always be equal to one.
 
-        p_values_arr = tmp_values_arr[idx].sum(axis=0)  # Just sum where TX >= TX0
+        # Just sum where TX >= TX0
+        p_values_arr = tmp_values_arr[idx].sum(axis=0)
         max_pvalue_index = p_values_arr.argmax()
 
-        inf_bound = (
-            nuisance_arr[0, 0, max_pvalue_index - 1]
-            if max_pvalue_index > 0
-            else nuisance_arr[0, 0, 0]
-        )
-        sup_bound = (
-            nuisance_arr[0, 0, max_pvalue_index + 1]
-            if max_pvalue_index < nuisance_num - 1
-            else nuisance_arr[0, 0, -1]
-        )
-
-    p_value = p_values_arr[max_pvalue_index]  # take the max value
-
-    if p_value > 1:
-        # Occurs because of numerical errors
-        p_value = 1.0
+        p_value = p_values_arr[max_pvalue_index]  # take the max value
+        if p_value > 1:
+            return 1
```

Comment:
I believe we could clip after the optimization no?

Reply:
Oh yes, I didn't know what `clip` was doing, but definitely